app/controllers/admin/preferences_controller.rb

```diff
@@ -0,0 +1,82 @@
+class Admin::PreferencesController < Admin::BaseController
+  before_action :check_ownership
+  before_action :check_totp_disabled, only: [:totp_setup, :totp_setup_form]
+
+  def check_ownership
+    admin_only_access_denied unless params[:admin_id] == current_admin.login
+  end
+
+  def check_totp_disabled
+    return unless current_admin.otp_required_for_login
+
+    flash[:error] = t("admin.preferences.check_totp_disabled.already_enabled")
+    redirect_to admins_path
+  end
+
+  def show
+    @totp_enabled = current_admin.otp_required_for_login
+
+    render "admin/preferences/show"
```

Comment:
This, and the ones below, can be removed since they match the default template path. (I wasn't sure because admin stuff is wonky, so I tried commenting them out in my dev environment.)